### ISSUE TITLE
Panic inside of an argument with a type-check shows a Type Error with no details in the IDE

### ISSUE THREAD

**Issue by radeusgd** [enso-org/enso#7837]
Repro:

Open the **IDE** and create node: `"a".replace 'foo\Lbar' ''`.

### Actual behaviour

This yields:
![image](https://github.com/enso-org/enso/assets/1436948/43801490-ffbe-426a-af38-de93def13496)

### Expected behaviour

1. I think the inner Panic should propagate outwards and _not_ be replaced with a type error instead. 
2. Regardless, the error message displayed must be human readable, not just `Type_Error.Error`.

### Context

Note that similar code in CLI simply fails at compile time with a syntax error:
```
from Standard.Base import all

main =
    "a".replace 'a\Lb' 'aa'
```
```
C:\NBO\repr\compiler-error.enso:4:17: error: Invalid escape sequence ab.
    4 |     "a".replace 'a\Lb' 'aa'
      |                 ^~~~~~
Aborting due to 1 errors and 0 warnings.
Execution finished with an error: Compilation aborted due to errors.
```

If the text is put as a standalone node in the IDE, it also displays the syntax error correctly:
![image](https://github.com/enso-org/enso/assets/1436948/c23a046b-f3cd-4185-b043-abbeb0e3fa71)

If I put a regular but not-erroneous value in this argument, it yields a good looking type error:
![image](https://github.com/enso-org/enso/assets/1436948/9b1dc59d-dc86-4d65-a561-f1693fa3fa0f)

Also, if I connect a standalone node to the `replace`, it also shows the buggy behaviour:
![image](https://github.com/enso-org/enso/assets/1436948/1214e557-01a3-46a1-bf5c-77da7f485ce6)

**Comment by radeusgd** [comment]
It seems most likely to be an engine issue, I'd suspect an issue with Panic sentinels. However the `Type_Error.Error` display issue - that's hard for me to say if it's in engine or IDE.

**Comment by radeusgd** [comment]
Note that this does not necessarily pertain to just syntax errors. It seems to be a problem with any panics.

See - when I throw a `42` as a panic value (weird but allowed). The node itself shows the panic OK:
![image](https://github.com/enso-org/enso/assets/1436948/ece97561-c16a-48e0-8bf8-6413c7203cbc)

But once I connect that to the `replace` node, it again is failing with the unclear `Type_Error`:
![image](https://github.com/enso-org/enso/assets/1436948/f4ac3d3b-81fc-452e-bd82-ca81f338150f)

Also happens if I inline it:
![image](https://github.com/enso-org/enso/assets/1436948/bfdae37e-937c-4a4d-af62-d40662b6758d)

I think the correct behaviour however, is to propagate the `42` panic. That is the behaviour of the analogous program in the CLI (the inlined version, as with multi lines CLI will obviously just stop execution at the Panic assignment):
```
from Standard.Base import all

main = "a".replace (Panic.throw 42) ''
```
```
Execution finished with an error: 42
        at <enso> Panic.throw(Internal)
        at <enso> compiler-error.main<arg-1>(compiler-error.enso:3:21-34)
        at <enso> compiler-error.main(compiler-error.enso:3:8-38)
```

**Comment by radeusgd** [comment]
It looks like this has been resolved by #7793, just the changes are not yet reflected in the cloud (checked moments ago), but on my local engine build the behaviour is already correct:

![image](https://github.com/enso-org/enso/assets/1436948/6d4590d6-8078-4edc-b384-f5ff7194da67)

So I assume the cloud will also get it once it updates to a newer nightly.
